pyrometheus/__init__.py

```diff
@@ -324,6 +410,76 @@ def get_temperature(self, H_or_E, T_guess, Y, do_energy=False):
 
         return T
 
+    def get_falloff_rates(self, T, C, k_fwd):
+        k_high = np.zeros(self.num_falloff)
+        k_low = np.zeros(self.num_falloff)
+        reduced_pressure = np.zeros(self.num_falloff)
+        falloff_function = np.zeros(self.num_falloff)
+
+         %for i, react in zip(range(len(falloff_reactions)), falloff_reactions):
+        k_high[${i}] = ${cgm(rate_coefficient_expr(react.high_rate, Variable("T")))}
+        k_low[${i}] = ${cgm(rate_coefficient_expr(react.low_rate, Variable("T")))}
+        reduced_pressure[${i}] = (${cgm(third_body_efficiencies_expr(
+            sol, react, Variable("C")))})*k_low[${i}]/k_high[${i}]
+        %if react.falloff.falloff_type == "Troe":
+        falloff_center = np.log10(
+            ${cgm(troe_falloff_expr(react, Variable("T")))})
+        log_rp = np.log10(reduced_pressure[${i}])
+        c = -0.4-0.67*falloff_center
+        n = 0.75-1.27*falloff_center
+        f = (log_rp+c)/(n-0.14*(log_rp+c))
+        falloff_function[${i}] = 10.0**(falloff_center/(1.0+f**2))
+        %else:
+        falloff_function[${i}] = 1.0
+        %endif
```

Comment:
I wish we could indent this somehow. But all the ways I can think of don't necessarily make it better. Have any ideas?

Reply:
I no longer compute falloff rates this way, so this bit of code is gone. This change is due to your subsequent comment about empty initialization.